PowerValidatedSolutions.psm1

```diff
@@ -50066,6 +50066,372 @@ Function Test-WMSubnetInput {
 }
 Export-ModuleMember -Function Test-WMSubnetInput
 
+Function Invoke-GeneratePrivateKeyAndCsrFiles {
+    <#
+        .SYNOPSIS
+        Generate Private Key and certificate signing request (CSR) files using Openssl with a common name and neccesary parameters.
+
+        .DESCRIPTION
+        The Invoke-GeneratePrivateKeyAndCsrFiles cmdlet generates the private key and certificate signing request (CSR) files using Openssl with a common name and neccesary parameters.
+        
+        .EXAMPLE
+        Invoke-GeneratePrivateKeyAndCsrFiles -certDir "C:\certificates" -commonName "sfo-vrli01.rainpole.io" -subjectAlternativeNames "sfo-vrli01a.rainpole.io, sfo-vrli01b.rainpole.io, sfo-vrli01c.rainpole.io" -keySize 4096 -expireDays 720 -organization "rainpole" -organizationUnit "IT" -locality "San Francisco" -state "California" -country "US"
+        This example will generate a private key and csr pair for a VIP sfo-vrli01.rainpole.io with cluster nodes sfo-vrli01a, sfo-vrli01b and sfo-vrli01c as SAN values.  
+
+        .PARAMETER certDir
```

Comment:
We use "path" is other spots but here use "dir".

Reply:
i have renamed it to certDirPath.  Instead of file path it is directory path so i kept the dir in the name.  
addressed in commit 9d50a853ffce55094985dea3d9e658670a7ab67a